File: app/src/main/java/se/appshack/android/refactoring/main/data/model/PokemonTypeModel.java
package se.appshack.android.refactoring.main.data.model;

import com.google.gson.annotations.SerializedName;

public class PokemonTypeModel {

    @SerializedName("slot")
    public int slot;

    @SerializedName("type")
    public NamedResponseModel type;
}

File: app/src/main/java/se/appshack/android/refactoring/main/data/model/PokemonTypeModel.kt
package se.appshack.android.refactoring.main.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class PokemonTypeModel (

    @SerializedName("slot")
    var slot: Int = 0,

    @SerializedName("type")
    var type: NamedResponseModel? = null

): Parcelable

